File: JBGE_RealityKit/JBGE_RealityKit/GameEngine/IDKeyboard.swift
//
//  IDKeyboard.swift
//  JBGE_RealityKit
//
//  Created by Tomohiro Kadono on 2026/01/04.
//

import Foundation

public class IDKeyboard {
    /// <summary>Detects multiple key presses</summary>
    public func IsKeyCombinationPressed(_ keyCode1: Int, _ keyCode2: Int) -> Bool {
        return Input.GetKey(keyCode1) && Input.GetKey(keyCode2)
    }

    /// <summary>Detects if the specified key is pressed</summary>
    public func IsKeyPressed(_ keyCode: Int) -> Bool {
        return Input.GetKey(keyCode)
    }
    
    public var PressedKeyCode: Int {
        get {
            return Input.FirstKey()
        }
    }

    public var IsAnyKeyPressed: Bool {
        get {
            return Input.AnyKey()
        }
    }
}

File: JBGE_Unity/Assets/JBGE_Unity/GameEngine/IDKeyboard.cs
using UnityEngine;
using UnityEngine.InputSystem;
using UnityEngine.InputSystem.Controls;

namespace JBGE {
	public class IDKeyboard {
    /// <summary>Detects multiple key presses</summary>
    /// <param name="keyCode1"></param>
    /// <param name="keyCode2"></param>
    /// <returns>true if specified keys are pressed together, else false</returns>
    public bool IsKeyCombinationPressed(int keyCode1, int keyCode2) {
      return Input.GetKey((KeyCode)keyCode1) && Input.GetKey((KeyCode)keyCode2);
    }

    /// <summary>Detects if the specified key is pressed</summary>
    /// <param name="keyCode"></param>
    /// <returns>true if specified key was pressed, else false</returns>
    public bool IsKeyPressed(int keyCode) {
      return Input.GetKey((KeyCode)keyCode);
    }

    public int PressedKeyCode {
			get {
          foreach(KeyControl kc in Keyboard.current.allKeys) {
            if(kc.isPressed) {
              return (int)kc.keyCode;
            }
          }
        return -1;
      }
		}

    public bool IsAnyKeyPressed {
      get {
        return Keyboard.current.anyKey.isPressed;
      }
    }
	}
}
